homeassistant/components/wled/switch.py

```diff
@@ -0,0 +1,174 @@
+"""Support for WLED switches."""
+import logging
+from typing import Any, Callable, List
+
+from wled import WLED, WLEDError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity, ToggleEntity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import WLEDDeviceEntity
+from .const import (
+    ATTR_DURATION,
+    ATTR_FADE,
+    ATTR_TARGET_BRIGHTNESS,
+    ATTR_UDP_PORT,
+    DATA_WLED_CLIENT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up WLED switch based on a config entry."""
+    wled: WLED = hass.data[DOMAIN][entry.entry_id][DATA_WLED_CLIENT]
+
+    switches = [
+        WLEDNightlightSwitch(entry.entry_id, wled),
+        WLEDSyncSendSwitch(entry.entry_id, wled),
+        WLEDSyncReceiveSwitch(entry.entry_id, wled),
+    ]
+    async_add_entities(switches, True)
+
+
+class WLEDSwitch(ToggleEntity, WLEDDeviceEntity):
```

Comment:
It doesnt matter now, but it's a bit more future proof to invert the inheritance order, so that the `WLEDDeviceEntity` is lower than the `ToggleEntity` class.

Speaking of ToggleEntity. It should be `SwitchDevice` from the switch integration that we inherit.

Reply:
Done.